tests/tests/smt_operation_tests.cpp

```diff
@@ -924,5 +924,110 @@ BOOST_AUTO_TEST_CASE( smt_limit_order_create2_apply )
+      op.add_reward_token( asset( 0, smt3 ) );
+
+      BOOST_TEST_MESSAGE( "Testing single reward claims" );
+      op.add_reward_token( ASSET( "1.000 TESTS" ) );
+      op.validate();
+      op.reward_tokens.clear();
+
+      op.add_reward_token( ASSET( "1.000 TBD" ) );
+      op.validate();
+      op.reward_tokens.clear();
+
+      op.add_reward_token( ASSET( "1.000000 VESTS" ) );
+      op.validate();
+      op.reward_tokens.clear();
+
+      op.add_reward_token( asset( 1, smt1 ) );
+      op.validate();
+      op.reward_tokens.clear();
+
+      op.add_reward_token( asset( 1, smt2 ) );
+      op.validate();
+      op.reward_tokens.clear();
+
+      op.add_reward_token( asset( 1, smt3 ) );
+      op.validate();
+      op.reward_tokens.clear();
+
+      BOOST_TEST_MESSAGE( "Testing invalid rewards" );
+      STEEM_REQUIRE_THROW( op.add_reward_token( ASSET( "-1.000 TESTS" ) ), fc::assert_exception );
+      STEEM_REQUIRE_THROW( op.add_reward_token( ASSET( "-1.000 TBD" ) ), fc::assert_exception );
+      STEEM_REQUIRE_THROW( op.add_reward_token( ASSET( "-1.000000 VESTS" ) ), fc::assert_exception );
+      STEEM_REQUIRE_THROW( op.add_reward_token( asset( -1, smt1 ) ), fc::assert_exception );
+      STEEM_REQUIRE_THROW( op.add_reward_token( asset( -1, smt2 ) ), fc::assert_exception );
+      STEEM_REQUIRE_THROW( op.add_reward_token( asset( -1, smt3 ) ), fc::assert_exception );
+      
+      BOOST_TEST_MESSAGE( "Testing inconsistencies of manually inserted reward tokens." );
+      op.reward_tokens.push_back( asset( 1, smt3 ) );
+      op.reward_tokens.push_back( asset( 1, smt1 ) );
+      STEEM_REQUIRE_THROW( op.validate(), fc::assert_exception );
+      op.reward_tokens.clear();
+      op.reward_tokens.push_back( asset( 1, smt3 ) );
+      op.reward_tokens.push_back( asset( 1, smt3 ) );
+      STEEM_REQUIRE_THROW( op.validate(), fc::assert_exception );
+      op.reward_tokens.clear();
+      op.reward_tokens.push_back( asset( -1, smt3 ) );
+      STEEM_REQUIRE_THROW( op.validate(), fc::assert_exception );
+      op.reward_tokens.clear();
+      op.reward_tokens.push_back( asset( 1, smt1 ) );
+      op.reward_tokens.push_back( asset( -1, smt3 ) );
+      STEEM_REQUIRE_THROW( op.validate(), fc::assert_exception );
+   }
+   FC_LOG_AND_RETHROW()
+}
+
+BOOST_AUTO_TEST_CASE( claim_reward_balance2_authorities )
+{
+   try
+   {
+      BOOST_TEST_MESSAGE( "Testing: decline_voting_rights_authorities" );
+
+      claim_reward_balance2_operation op;
+      op.account = "alice";
+
+      flat_set< account_name_type > auths;
+      flat_set< account_name_type > expected;
+
+      op.get_required_owner_authorities( auths );
+      BOOST_REQUIRE( auths == expected );
+
+      op.get_required_active_authorities( auths );
+      BOOST_REQUIRE( auths == expected );
+
+      expected.insert( "alice" );
+      op.get_required_posting_authorities( auths );
+      BOOST_REQUIRE( auths == expected );
+   }
+   FC_LOG_AND_RETHROW()
+}
+
```

Comment:
There is no `claim_reward_balance2_apply` test. At the very least, it should contain the identical test for STEEM, SBD, and VESTS as `claim_reward_balance_apply` to verify existing behavior is not broken.

Reply:
No, there is no evaluation tests because as far as I know both evaluation and its testing were outside the scope of the original issue (which was https://github.com/steemit/steem/issues/1896 ). That was brought to my attention after I had already had the validator implemented so I decided to leave it as is and created original PR (which was https://github.com/steemit/steem/pull/1997 ). Then @theoreticalbts decided to create another issue (for reasons unknown to me) and so we're here. Shall we expand this issue/PR to create the evaluation test too, I will.

Reply:
@youkaicountry Your call.

Reply:
However, because there is an implemented evaluator, I expect it to be tested. If the evaluator is out of scope of this issue, then it should be removed from this PR and added to another.

It is too easy to see the implemented evaluator and assume the other issue is complete when it is not because of a lack of a test.

Reply:
I don't want to delay merging of this PR, but I don't want to merge blatantly untested code.

Reply:
@mvandeberg @youkaicountry I committed `claim_reward_balance2_apply` test that contains both legacy symbols and SMT testing copying in scope what original `claim_reward_balance_apply` does. Now that the `claim_reward_balance2_evaluator` is being tested, this PR is ready to be merged, pending CI pass.